src/test/kotlin/com/github/avrokotlin/avro4k/endecode/ArrayEncoderTest.kt

```diff
@@ -0,0 +1,51 @@
+package com.github.avrokotlin.avro4k.endecode
+
+import com.github.avrokotlin.avro4k.record
+import io.kotest.core.factory.TestFactory
+import io.kotest.core.spec.style.WordSpec
+import io.kotest.core.spec.style.wordSpec
+import kotlinx.serialization.Serializable
+
+class ArrayEncoderTest : WordSpec({
+   includeForEveryEncoder { arrayEncodingTests(it) }
+})
+
+fun arrayEncodingTests(encoderToTest: EnDecoder): TestFactory {
+   return wordSpec {
+      "en-/decoder" should {
+         "generate GenericData.Array for an Array<Boolean>" {
+            @Serializable
+            data class ArrayBooleanTest(val a: Array<Boolean>)
+            val value = ArrayBooleanTest(arrayOf(true, false, true))
+            encoderToTest.testEncodeDecode(value, record(value.a.asList()))
+         }
+         "support GenericData.Array for an Array<Boolean> with other fields" {
```

Comment:
For these tests that expect the same as the other but including other fields, maybe we could have a big data class with all the possible types (null/non-null, raw types, generics, enums, arrays, nested data classes, ...), it would naturally check that encoding and decoding are working even with complex objects and other fields. We would re-use the same as what I said just before (e.g. having a test method that handle both the data class instance and its avro representation).

Also, it would act as a full example of what can do the avro4k library.

Reply:
This is a good additional test case, but I would rather leave the specific test cases also in our test suite.